models/search/SearchActiveRecord.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+namespace app\models\search;
+
+
+use Yii;
+use yii\base\Exception;
+use yii\helpers\Inflector;
+
+abstract class SearchActiveRecord extends \yii\elasticsearch\ActiveRecord
+{
+    /**
+     * @var array language analyzers for elasticsearch based on
+     * https://www.elastic.co/guide/en/elasticsearch/reference/5.6/analysis-lang-analyzer.html
+     */
+    public static $languages = [
+        // lang => analyzer name
+        'en' => 'english',
+        'de' => 'german',
+        'es' => 'spanish',
+        'fr' => 'french',
+        'he' => 'standard', // 'hebrew' // currently not supported by elasticsearch
+        'id' => 'indonesian',
+        'it' => 'italian',
+        'ja' => 'standard', // 'japanese' // currently not supported by elasticsearch
+        'pl' => 'standard', // 'polish' // currently not supported by elasticsearch
+        'pt' => 'portuguese',
+        'pt-br' => 'portuguese',
+        'ro' => 'romanian',
+        'ru' => 'russian',
+        'sv' => 'swedish',
+        'uk' => 'russian', // 'ukrainian', // currently not supported by elasticsearch
+        'zh-cn' => 'chinese',
+    ];
+
+
+    public static function index()
+    {
+        return YII_ENV_DEV ? 'yiiframework-dev' : 'yiiframework';
+    }
+
+    /**
+     * @inheritdoc
+     * @return SearchActiveQuery the newly created [[ActiveQuery]] instance.
+     */
+    public static function find()
+    {
+        return new SearchActiveQuery(get_called_class());
+    }
+
+    public static function search($queryString, $version = null, $language = null, $type = null)
+    {
+        $query = static::find();
+
+        if (isset(static::$languages[$language])) {
+            $indexes = [static::index() . "-$language"];
+            $analyzer = static::$languages[$language];
+        } else {
+            $language = null;
+            $indexes = array_map(function($i) { return static::index() . "-$i"; }, array_keys(static::$languages));
+            $analyzer = 'english';
+            $query->indicesBoost = [
+                static::index() . '-en' => 20,
+            ];
+        }
+
+        $types = [];
+        if ($type === null) {
+            $types = ['api-type',/* 'api-primitive',*/ 'guide-section', 'wiki', 'extension', 'news'];
+        } elseif (in_array($type, ['wiki', 'extension', 'news'], true)) {
+            $types = [$type];
+        } elseif ($type === 'api') {
+            $types = ['api-type',/* 'api-primitive',*/];
+        } elseif ($type === 'guide') {
+            $types = ['guide-section']; // TODO add possibility to search for guide subsections
```

Comment:
Aren't they indexed?

Reply:
currently a search result item will refer to a whole guide page, it may be nice to display subsections as separate items. See https://github.com/yiisoft-contrib/yiiframework.com/issues/228